### PR TITLE
Introduce dependency on solidity-multicall 

### PR DESCRIPTION
**Pull request by bh2smith**
Now using a more generic utility for testing Full batches! Thanks @fleupold for showing me some sweet pointers.

### CODE REVIEW

**Review by josojo — APPROVED** [review]
sweet. neat package!

File: test/snapp_base.js [hunk 1]
```diff
@@ -998,12 +996,17 @@ contract("SnappBase", async (accounts) => {
 
   describe("Larger Test Cases", () => {
     it("Fill deposit batch", async () => {
-      const instance = await MultiFlux.new()
+      const multiCaller = await MultiCaller.new()
+      const instance = await SnappBase.new()
       const core = await SnappBaseCore.new()
-      await setupEnvironment(MintableERC20, instance, token_owner, [user_1], 2)
+      // Note that only the token at index 1 is important here.
+      const token = (await setupEnvironment(MintableERC20, instance, token_owner, [user_1], 2))[1]
```
Comment (fleupold) [inline test/snapp_base.js]: nit: I'd just take all tokens here and register them for convenience (instead of just taking the 2nd token).

**Review by fleupold — APPROVED** [review]
Awesome 

🥫

File: test/snapp_base.js (hunk 1 above)
Comment (bh2smith) [reply test/snapp_base.js]: done!

**Pull request merged**